NetKet/Graph/custom_graph.hpp

```diff
@@ -179,15 +82,100 @@ class CustomGraph : public AbstractGraph {
 
   // Returns map of the edge and its respective color
   const ColorMap &EdgeColors() const override { return eclist_; }
+};
 
- private:
-  bool ComputeConnected() const {
-    const int start = 0;  // arbitrary node
-    int nvisited = 0;
-    BreadthFirstSearch(start, [&nvisited](int, int) { ++nvisited; });
-    return nvisited == Nsites();
+// TODO(twesterhout): Move me somewhere :)
+inline std::vector<std::vector<int>> AdjacencyListFromEdges(
+    const std::vector<std::vector<int>> &edges) {
+  std::vector<std::vector<int>> adjlist;
+  auto nsites = 0;
+
+  for (auto const &edge : edges) {
+    if (edge.size() != 2) {
+      throw InvalidInputError(
+          "The edge list is invalid (edges need "
+          "to connect exactly two sites)");
+    }
+    if (edge[0] < 0 || edge[1] < 0) {
+      throw InvalidInputError("The edge list is invalid");
+    }
+
+    nsites = std::max(std::max(edge[0], edge[1]), nsites);
   }
-};
+
+  nsites++;
+  adjlist.resize(static_cast<std::size_t>(nsites));
+
+  for (auto const &edge : edges) {
+    adjlist[edge[0]].push_back(edge[1]);
+    adjlist[edge[1]].push_back(edge[0]);
+  }
+  return adjlist;
+}
+
+// TODO(twesterhout): Move me to json_helpers.hpp or something and make into a
+// normal function (i.e. not a template) :)
```

Comment:
No, there's just no more reason for accepting `pybind11::kwargs` anymore, since we've already handled construction in pynetket.cc. And then it's always nicer to have concrete functions.